File: Python/lista_de_exercicios_01/classes/imprimiveis.py
from abc import ABC, abstractmethod


class Imprimivel(ABC):
    @abstractmethod
    def imprimir(self):
        pass


class Relatorio(Imprimivel):
    def __init__(self, nome, texto):
        self.nome = nome
        self.texto = texto

    def imprimir(self):
        print("Imprimindo relatório")


class Contrato(Imprimivel):
    def __init__(self, nome, partes, termos):
        self.nome = nome
        self.partes = partes
        self.termos = termos

    def imprimir(self):
        print("Imprimindo contrato")

File: Golang/lista_de_exercicios_01/classes/imprimiveis/imprimiveis.go
package imprimiveis

import "fmt"

type Imprimivel interface {
	Imprimir()
}

type Contrato struct {
	Nome   string
	Partes string
	Termos string
}

type Relatorio struct {
	Nome  string
	Texto string
}

func NewContrato(nome string, partes string, termos string) Contrato {
	c := Contrato{Nome: nome, Partes: partes, Termos: termos}
	return c
}

func NewRelatorio(nome string, texto string) Relatorio {
	r := Relatorio{Nome: nome, Texto: texto}
	return r
}

func (c Contrato) Imprimir() {
	fmt.Println("Função de imprimir da interface Imprimível em Contrato")
}

func (r Relatorio) Imprimir() {
	fmt.Println("Função de imprimir da interface Imprimível em Relatório")
}
